### ISSUE TITLE
Non-Deterministic Behavior in BertTokenizer

### ISSUE THREAD

**Issue by 4tywon** [huggingface/transformers#2306]
## 🐛 Bug

Model I am using (Bert, XLNet....): BertTokenizer only

Language I am using the model on (English, Chinese....):

The problem arise when using:
* [ ] the official example scripts: (give details)
* [X] my own modified scripts: Jupyter Notebook 

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [X] my own task or dataset: (give details) tokenize a string

## To Reproduce
Jupyter Notebook with one cell, with a cloned version of the transformers repo.

```
import sys
sys.path.insert(0, 'transformers')

from transformers import BertTokenizer

tokenizer = BertTokenizer.from_pretrained('bert-base-uncased', sep_token='[SEP]', do_lower_case=True)

tokenizer.tokenize("[PAD] [SEP] [SEP] [PAD]")
```

with outputs (varying between kernel restarts and runs):
`['[PAD]', '[', 'sep', ']', '[', 'sep', ']', '[PAD]']`
`['[PAD]', '[SEP]', '[SEP]','[PAD]']`
`['[PAD]', '[SEP]', '[SEP]', '[', 'pad', ']']`
## Expected behavior
Expected the output to be 
`['[PAD]', '[SEP]', '[SEP]','[PAD]']`
and have deterministic behavior

## Environment

Platform Linux-3.10.0-1062.el7.x86_64-x86_64-with-redhat-7.7-Verona
Python 3.7.4 (default, Aug 13 2019, 20:35:49) 
[GCC 7.3.0]
PyTorch 1.3.1
Tensorflow 1.15.0

* PyTorch Transformers version (or branch): on master branch of repo
* Using GPU ? Yes 

## Additional context

This happens sometimes when the notebook kernel is restarted and the cell is re-run. I haven't observed this happening when running a python script.

**Comment by 4tywon** [comment]
As an update I've been able to reproduce this problem using a python script

**Comment by BramVanroy** [comment]
I cannot reproduce this on Windows, PT1.4, latest transformers master.

**Comment by 4tywon** [comment]
I think I was using an old version of transformers :( 

This seems to have been fixed in v2.2.2 - After upgrading to latest I haven't observed this anymore so I'll close this issue.

Thanks!
